File: ios/ExpoNativeWechatModuleParams.swift
//
//  ExpoNativeWechatModuleParams.swift
//  ExpoNativeWechat
//
//  Created by Hector Chong on 4/7/24.
//

import ExpoModulesCore

struct BasicRequestParams: Record {
    @Field
    var id: String
}

struct RegisterAppParams: Record {
    @Field
    var id: String
    
    @Field
    var log: Bool = false
    
    @Field
    var appid: String
    
    @Field
    var logPrefix: String?
    
    @Field
    var universalLink: String
}

struct SendAuthRequestParams: Record {
    @Field
    var id: String
    
    @Field
    var state: String
    
    @Field
    var scope: String?
}


struct ShareTextParams: Record {
    @Field
    var id: String
    
    @Field
    var text: String
    
    @Field
    var scene: Int
}


struct ShareImageParams: Record {
    @Field
    var id: String
    
    @Field
    var src: String
    
    @Field
    var scene: Int
}

struct ShareVideoParams: Record {
    @Field
    var id: String
    
    @Field
    var videoUrl: String
    
    @Field
    var title: String?
    
    @Field
    var description: String?
    
    @Field
    var videoLowBandUrl: String?
    
    @Field
    var coverUrl: String?
    
    @Field
    var scene: Int
}

struct ShareWebpageParams: Record {
    @Field
    var id: String
    
    @Field
    var webpageUrl: String
    
    @Field
    var title: String?
    
    @Field
    var description: String?
    
    @Field
    var coverUrl: String?
    
    @Field
    var scene: Int
}

struct ShareMiniProgramParams: Record {
    @Field
    var id: String
    
    @Field
    var userName: String
    
    @Field
    var miniProgramType: Int
    
    @Field
    var webpageUrl: String
    
    @Field
    var path: String
    
    @Field
    var withShareTicket: Bool?
    
    @Field
    var title: String?
    
    @Field
    var description: String?
    
    @Field
    var coverUrl: String?
}

struct RequestPaymentParams: Record {
    @Field
    var id: String
    
    @Field
    var partnerId: String
    
    @Field
    var nonceStr: String
    
    @Field
    var prepayId: String
    
    @Field
    var timeStamp: String
    
    @Field
    var sign: String
}

struct RequestSubscribeMsgParams: Record {
    @Field
    var id: String
    
    @Field
    var scene: Int
    
    @Field
    var templateId: String
    
    @Field
    var reserved: String?
}

struct LaunchMiniprogramParams: Record {
    @Field
    var id: String
    
    @Field
    var userName: String
    
    @Field
    var path: String
    
    @Field
    var miniProgramType: Int
}

struct OpenCustomerServiceParams: Record {
    @Field
    var id: String
    
    @Field
    var corpid: String
    
    @Field
    var url: String
}

File: android/src/main/java/expo/modules/nativewechat/ExpoNativeWechatModuleParams.kt
package expo.modules.nativewechat

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class BasicRequestParams : Record {
  @Field
  val id: String = ""
}

class RegisterAppParams : Record {
  @Field
  val id: String = ""

  @Field
  val log: Boolean = false

  @Field
  val appid: String = ""

  @Field
  val logPrefix: String? = null

  @Field
  val universalLink: String = ""
}

class SendAuthRequestParams : Record {
  @Field
  val id: String = ""

  @Field
  val state: String = ""

  @Field
  val scope: String? = null
}


class ShareTextParams : Record {
  @Field
  val id: String = ""

  @Field
  val text: String = ""

  @Field
  val scene: Int = 0
}


class ShareImageParams : Record {
  @Field
  val id: String = ""

  @Field
  val src: String = ""

  @Field
  val scene: Int = 0
}

class ShareVideoParams : Record {
  @Field
  val id: String = ""

  @Field
  val videoUrl: String = ""

  @Field
  val title: String? = null

  @Field
  val description: String? = null

  @Field
  val videoLowBandUrl: String? = null

  @Field
  val coverUrl: String? = null

  @Field
  val scene: Int = 0
}

class ShareWebpageParams : Record {
  @Field
  val id: String = ""

  @Field
  val webpageUrl: String = ""

  @Field
  val title: String? = null

  @Field
  val description: String? = null

  @Field
  val coverUrl: String? = null

  @Field
  val scene: Int = 0
}

class ShareMiniProgramParams : Record {
  @Field
  val id: String = ""

  @Field
  val userName: String = ""

  @Field
  val miniProgramType: Int = 0

  @Field
  val webpageUrl: String = ""

  @Field
  val path: String = ""

  @Field
  val withShareTicket: Boolean? = null

  @Field
  val title: String? = null

  @Field
  val description: String? = null

  @Field
  val coverUrl: String? = null

  @Field
  val scene: Int = 0
}

class RequestPaymentParams : Record {
  @Field
  val id: String = ""

  @Field
  val partnerId: String = ""

  @Field
  val nonceStr: String = ""

  @Field
  val prepayId: String = ""

  @Field
  val timeStamp: String = ""

  @Field
  val sign: String = ""

  @Field
  val extData: String = ""
}

class RequestSubscribeMsgParams : Record {
  @Field
  val id: String = ""

  @Field
  val scene: Int = 0

  @Field
  val templateId: String = ""

  @Field
  val reserved: String? = null
}

class LaunchMiniprogramParams : Record {
  @Field
  val id: String = ""

  @Field
  val userName: String = ""

  @Field
  val path: String = ""

  @Field
  val miniProgramType: Int = 0
}

class OpenCustomerServiceParams : Record {
  @Field
  val id: String = ""

  @Field
  val corpid: String = ""

  @Field
  val url: String = ""
}
